catalog/tests/dags/providers/provider_api_scripts/test_science_museum.py

```diff
@@ -107,6 +107,56 @@ def test_get_query_param_offset_page_number():
     assert actual_param == expected_param
 
 
+@pytest.mark.parametrize(
+    "record",
+    [
+        # missing foreign_landing_url
+        {"links": {}},
+        {"links": {"self": ""}},
+        {"links": {"self": "link"}, "attributes": {"multimedia": []}},
+    ],
+)
+def test_get_record_data_returns_none_for_falsy_fid_and_landing_url(record):
```

Comment:
I didn't immediately understand why these cases are pulled out from the test case below. It's because when the foreign landing url or id are missing, we return `None` right away, but otherwise we attempt to build and return an image list -- which will be empty if the records were all invalid (missing image_url, etc). Could you add a comment to that effect? Or alternatively, I suppose they could all be in one test with `assert not actual_record_data`.